daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2083,17 +2087,16 @@ func (devices *DeviceSet) deleteDevice(info *devInfo, syncDelete bool) error {
 func (devices *DeviceSet) DeleteDevice(hash string, syncDelete bool) error {
 	logrus.Debugf("devmapper: DeleteDevice START(hash=%v syncDelete=%v)", hash, syncDelete)
 	defer logrus.Debugf("devmapper: DeleteDevice END(hash=%v syncDelete=%v)", hash, syncDelete)
-	info, err := devices.lookupDeviceWithLock(hash)
+	devices.Lock()
+	defer devices.Unlock()
+	info, err := devices.lookupDevice(hash)
```

Comment:
lock ordering is info lock and then devices lock. Can't hold devices lock and then take info lock. That can lead to deadlock.